ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_creation.py

```diff
@@ -302,3 +302,58 @@ def test_paddle_empty(
         shape=shape,
         dtype=dtype[0],
     )
+
+
+@st.composite
+def _custom_diag_helper(draw):
+    dtype, x = draw(
+        helpers.dtype_and_values(
+            available_dtypes=helpers.get_dtypes("numeric"),
+            small_abs_safety_factor=2,
+            large_abs_safety_factor=2,
+            safety_factor_scale="log",
+            min_num_dims=1,
+            max_num_dims=2,
+            min_dim_size=1,
+            max_dim_size=50,
+        )
+    )
+    shape = x[0].shape
+    if len(shape) == 2:
+        k = draw(helpers.ints(min_value=-shape[0] + 1, max_value=shape[1] - 1))
+    else:
+        k = draw(helpers.ints(min_value=0, max_value=shape[0]))
+    # p = draw(helpers.number(min_value=0, max_value=50))
+    # return dtype, x, k, p
+    return dtype, x, k
+
+
+# diag
+@handle_frontend_test(
+    fn_tree="paddle.diag",
+    # dtype_and_x_k_p=_custom_diag_helper(),
+    dtype_and_x_k=_custom_diag_helper(),
+    # p=helpers.number(min_value=0, max_value=50)
+    p=st.floats(0, 1),
```

Comment:
You can combine multiple strategies using the `st.one_of(..)` strategy. Include the strategies for generating integers `helpers.ints(...)` and `helpers.floats(...)` inside `st.one_of` for pickling randomly from either of these strategies and drawing the test data from that on runtime